lib/infra/infra-stack.ts

```diff
@@ -540,28 +555,31 @@ export class InfraStack extends Stack {
 
       if (managerAsgCapacity > 0) {
         const managerNodeAsg = new AutoScalingGroup(this, 'managerNodeAsg', {
-          vpc: props.vpc,
-          instanceType: defaultInstanceType,
-          machineImage: MachineImage.latestAmazonLinux2023({
-            cpuType: instanceCpuType,
+          launchTemplate: new LaunchTemplate(this, 'managerNodeLt', {
+            instanceType: defaultInstanceType,
+            machineImage: MachineImage.latestAmazonLinux2023({
+              cpuType: instanceCpuType,
+            }),
+            role: this.instanceRole,
+            securityGroup: props.securityGroup,
+            blockDevices: [{
+              deviceName: '/dev/xvda',
+              volume: BlockDeviceVolume.ebs(50, { deleteOnTermination: true, volumeType: props.storageVolumeType }),
+            }],
+            requireImdsv2: true,
+            userData: UserData.forLinux(),
```

Comment:
confused about forLinux() here. Looks like userdata is optional right? https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.LaunchTemplate.html#userdata and all installations we are running via cfnInit?

Reply:
Not optional when using LaunchTemplate, this needs to be declared. This adds below:
```
UserData:
          Fn::Base64:
            Fn::Join:
              - ""
              - - |-
                  #!/bin/bash
                  # fingerprint: 1ef4754d2d6b55e6
                  (
                    set +e
                    /opt/aws/bin/cfn-init -v --region
                - Ref: AWS::Region
                - " --stack "
                - Ref: AWS::StackName
                - |-2
                   --resource dataNodeAsgASGA92884EE -c default
                    /opt/aws/bin/cfn-signal -e $? --region
                - Ref: AWS::Region
                - " --stack "
                - Ref: AWS::StackName
                - |-2
                   --resource dataNodeAsgASGA92884EE
                    cat /var/log/cfn-init.log >&2
                  )
                  
```
This was getting added previously as part of Launch Configuration.